### PR TITLE
fix extraction of fragments (multiple named exports)

### PR DESCRIPTION
**Pull request by pieh**
https://github.com/gatsbyjs/gatsby/pull/6765 had flawed implementation that I didn't catch in review - it caused to only handle single fragment exported from file, making fragments exported by `gatsby-transformer-sharp` not usable - this fixes it.

It uses concatenated location of graphql template literal location inside source file and graphql document (query/fragment) location inside graphql template literal to determine uniqueness of returned graphql documents

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview for *using-postcss-sass* failed.

Built with commit ae161bc4f2b9f570c60ccb14158c23b2cbaa017c

https://app.netlify.com/sites/using-postcss-sass/deploys/5b6047c13813f003aa06dcf2

File: packages/gatsby/src/internal-plugins/query-runner/file-parser.js [hunk 1]
```diff
@@ -83,6 +83,15 @@ async function findGraphQLTags(file, text): Promise<Array<DefinitionNode>> {
           return
         }
 
+        /**
+         * Mapping of graphql documents to concatenated location of graphql template
+         * literal that contains that document and location of graphql document inside
+         * template literal.
```
Comment (m-allanson) [inline packages/gatsby/src/internal-plugins/query-runner/file-parser.js]: Could we make this comment simpler to understand? Maybe this line could be split into a _What?_ and then a _How?_ section?

> A map of graphql documents to unique locations.
>
> A graphql document's unique location is made of: 
>  - the location of the graphql template literal that contains the document, and
>  - the document's location within the graphql template literal

File: packages/gatsby/src/internal-plugins/query-runner/file-parser.js [hunk 2]
```diff
@@ -83,6 +83,15 @@ async function findGraphQLTags(file, text): Promise<Array<DefinitionNode>> {
...
+         *
+         * This is used to prevent returning duplicated documents.
```
Comment (m-allanson) [inline packages/gatsby/src/internal-plugins/query-runner/file-parser.js]: 👍 then this final line is the _Why?_

**Comment by pieh** [comment]
@m-allanson Thanks for making `documentLocations` explanation clearer!

**Pull request merged**